test-integration/helpers/calendar.coffee

```diff
@@ -1,6 +1,85 @@
 selenium = require 'selenium-webdriver'
 {expect} = require 'chai'
 
+{TestHelper} = require './test-element'
+
+COMMON_ELEMENTS =
+  forecastLink:
+    locator:
+      linkText: 'Performance Forecast'
+  addToggle:
+    locator:
+      css: '.add-assignment .dropdown-toggle'
+  addReadingButton:
+    locator:
+      linkText: 'Add Reading'
+  addHomeworkButton:
+    locator:
+      linkText: 'Add Homework'
+  addExternalButton:
+    locator:
+      linkText: 'Add External Assignment'
+
+
+COMMON_POPUP_ELEMENTS =
+  closeButton:
+    locator:
+      css: '.plan-modal .close'
+  editLink:
+    locator:
+      linkText: 'Edit Assignment'
+  reviewLink:
+    locator:
+      linkText: 'Review Metrics'
+
+class PlanPopupHelper extends  TestHelper
+  constructor: (test, testElementLocator) ->
+
+    testElementLocator ?= '.plan-modal .panel.panel-default'
+    super(test, testElementLocator, COMMON_POPUP_ELEMENTS)
+
+  close: =>
+    @getCloseButton().click()
+    @test.sleep(2000)
+
+  goEdit: =>
+    @getEditLink().click()
+
+  goReview: =>
+    @getReviewLink().click()
+
+class CalendarHelper extends TestHelper
+  constructor: (test, testElementLocator) ->
+
+    testElementLocator ?= '.calendar-container'
+    super(test, testElementLocator, COMMON_ELEMENTS, loadingLocator: '.calendar-loading')
+    @setCommonHelper('planPopup', new PlanPopupHelper(@test))
+
+  createNew: (type) =>
+    @waitUntilLoaded()
+
+    @getAddToggle().click()
+
+    switch type
+      when 'READING' then @getAddReadingButton().click()
+      when 'HOMEWORK' then @getAddHomeworkButton().click()
+      when 'EXTERNAL' then @getAddExternalButton().click()
+      else expect(false, 'Invalid assignment type').to.be.true
+
+  goPerformanceForecast: =>
+    @getForecastLink().click()
+
+  goOpen: (title) =>
+    # wait until the calendar is open
+    @waitUntilLoaded()
+    # TODO: Make this a `data-title` attribute
+    # HACK: Might need to scroll the item to click on into view
+    el = @test.waitAnd(css: "[data-title='#{title}']")
+    @test.scrollTo(el)
```

Comment:
`describe` already has a scrollToAndClick method (don't remember the exact name, maybe it's `waitClick`?) can you use that instead?